File: rag/src/api/schemas/conversation.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from rag.src.models.message import MessageRole


class MessageRead(BaseModel):
    id: uuid.UUID
    role: MessageRole
    content: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ConversationRead(BaseModel):
    id: uuid.UUID
    agent_id: uuid.UUID
    user_id: uuid.UUID
    created_at: datetime
    messages: list[MessageRead]

    model_config = {"from_attributes": True}

File: web/src/api/types/conversation.ts
import type { MessageRole } from '@/types/chat';

export type MessageRead = {
  id: string;
  role: MessageRole;
  content: string;
  created_at: string;
};

export type ConversationRead = {
  id: string;
  agent_id: string;
  user_id: string;
  created_at: string;
  messages: MessageRead[];
};

export type ConversationsListParams = {
  page?: number;
  limit?: number;
};